### ISSUE TITLE
[Question] Home directory is always being detected as project

### ISSUE THREAD

**Issue by maxpollack** [wsdjeg/SpaceVim#3319]
## Expected behavior, english is required
When opening a file in vim from a working directory that contains a `.git` folder, that folder will be treated as the project root and `fzf` behavior (as well as things like the runner) will be executed in the context of that working directory.

## The reproduce ways from Vim starting (Required!)
Any time I open any file that's saved in any subdirectory of my home directory, it will default to the entire home directory being my default project. This makes all the project-level searching/grepping useless as it greps every file on my machine.

I see that it's changing my working directory to my home directory in the logs you asked me to post, but I really don't have insight as to why. I do not have a `.git` folder in my home directory, and my other machines have a `.SpaceVim.d` without any trouble.

## Environment Information

- OS: linux
- vim version: -
- neovim version: 0.4.3
- SpaceVim version: 1.4.0-dev
- SpaceVim status: 

```
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

<details><summary> SpaceVim debug information </summary>

### SpaceVim options :

```toml
  auto_disable_touchpad = 1
  autocomplete_method = 'asyncomplete'
  autocomplete_parens = 1
  buffer_index_type = 4
  checkinstall = 1
  colorscheme = 'onedark'
  colorscheme_bg = 'dark'
  colorscheme_default = 'desert'
  commandline_prompt = '➭'
  custom_color_palette = []
  custom_plugins = []
  debug_level = 1
  default_indent = 2
  disabled_plugins = []
  enable_ale = 0
  enable_bepo_layout = 0
  enable_cursorcolumn = 0
  enable_cursorline = 1
  enable_debug = 0
  enable_googlesuggest = 0
  enable_guicolors = 1
  enable_insert_leader = 1
  enable_javacomplete2_py = 0
  enable_key_frequency = 0
  enable_language_specific_leader = 1
  enable_neocomplcache = 0
  enable_neomake = 1
  enable_os_fileformat_icon = 0
  enable_powerline_fonts = 1
  enable_statusline_bfpath = 0
  enable_statusline_mode = 1
  enable_statusline_tag = 1
  enable_tabline_filetype_icon = 1
  enable_tabline_ft_icon = 0
  enable_vimfiler_filetypeicon = 0
  enable_vimfiler_gitstatus = 0
  enable_vimfiler_welcome = 1
  enable_ycm = 0
  error_symbol = '✖'
  expand_tab = 1
  filemanager = 'vimfiler'
  filetree_direction = 'right'
  filetype_icons = {}
  force_global_config = 0
  gitcommit_issue_icon = ''
  gitcommit_pr_icon = ''
  github_username = ''
  guifont = ''
  hiddenfileinfo = 1
  home_files_number = 6
  hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'
  info_symbol = 'ⓘ'
  keep_server_alive = 1
  language = ''
  leader_guide_default_group_name = ''
  leader_guide_flatten = 1
  leader_guide_hspace = 5
  leader_guide_max_size = 0
  leader_guide_position = 'botright'
  leader_guide_run_map_on_popup = 1
  leader_guide_sort_horizontal = 0
  leader_guide_submode_mappings = {'<C-C>': 'win_close'}
  leader_guide_vertical = 0
  lint_on_save = 1
  lint_on_the_fly = 0
  max_column = 120
  plugin_bundle_dir = '/home/max/.cache/vimfiles/'
  plugin_manager = 'dein'
  plugin_manager_processes = 16
  plugin_name = 'goyo.vim'
  project_rooter_automatically = 1
  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/', 'Cargo.toml', '.SpaceVim.d/']
  realtime_leader_guide = 1
  relativenumber = 0
  search_tools = ['rg', 'ag', 'pt', 'ack', 'grep', 'findstr']
  sidebar_direction = ''
  sidebar_width = 30
  smartcloseignoreft = ['tagbar', 'vimfiler', 'defx', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']
  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
  snippet_engine = 'neosnippet'
  src_root = 'E:\sources\'
  statusline_inactive_separator = 'bar'
  statusline_iseparator = 'nil'
  statusline_left_sections = ['winnr', 'filename', 'major mode', 'search count', 'syntax checking', 'minor mode lighters']
  statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']
  statusline_separator = 'slant'
  statusline_unicode_symbols = 1
  terminal_cursor_shape = 2
  update_retry_cnt = 3
  version = '1.4.0-dev'
  vim_help_language = 'en'
  vimcompatible = 0
  warning_symbol = '⚠'
  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
  windows_index_type = 3
  windows_leader = 's'
  windows_smartclose = 'q'
```

### SpaceVim layers :

```toml
[[layers]]
  name="autocomplete"
  auto-completion-return-key-behavior='complete'
  auto-completion-tab-key-behavior='cycle'
[[layers]]
  name="checkers"
[[layers]]
  name="format"
[[layers]]
  name="edit"
[[layers]]
  name="ui"
[[layers]]
  name="core"
[[layers]]
  name="core#banner"
[[layers]]
  name="core#statusline"
[[layers]]
  name="core#tabline"
[[layers]]
  name="colorscheme"
[[layers]]
  name="shell"
  default_position='top'
  default_height=30
[[layers]]
  name="lang#go"
[[layers]]
  name="lang#rust"
  recommended-style=1
[[layers]]
  name="lang#ruby"
[[layers]]
  name="lsp"
[[layers]]
  name="fzf"
[[layers]]
  name="tools"
```

### SpaceVim Health checking :

SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim environment check report:
Current progpath: nvim(/usr/bin/nvim)
version: 800
OS: linux
[shell, shellcmdflag, shellslash]: ['/usr/bin/zsh', '-c', 0]
SpaceVim lua support check report:
Checking +lua:
      Failed : Known issue, neovim do not support lua now.
SpaceVim python support check report:
Checking +python3:
      Failed : to support +python3, you need run `pip3 install neovim`
Checking +python:
      Failed : to support +python, you need run `pip2 install neovim`

### SpaceVim runtime log :

```log
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!
[ SpaceVim ] [14:28:02] [ Info ] Startup with no argv, current dir is used: ~/dev/rust/test_app
[ SpaceVim ] [14:28:02] [ Info ] Can not find project local config, start to loadding global config
[ SpaceVim ] [14:28:02] [ Info ] start to apply config [glob]
[ SpaceVim ] [14:28:02] [ Info ] SpaceVim server startup at:/tmp/spacevim_nvim_server
[ SpaceVim ] [14:28:03] [ Info ] Start to find root for: /home/max/dev/rust/test_app/
[ SpaceVim ] [14:28:03] [ Info ]         (.git/):/home/max/dev/rust/test_app
[ SpaceVim ] [14:28:03] [ Info ]         (Cargo.toml):/home/max/dev/rust/test_app
[ SpaceVim ] [14:28:03] [ Info ]         (.SpaceVim.d/):/home/max
[ SpaceVim ] [14:28:03] [ Info ] buffer name: 
[ SpaceVim ] [14:28:03] [ Info ] change to root: /home/max
[ SpaceVim ] [14:28:03] [ Info ] try to open SpaceVim welcome page
[ SpaceVim ] [14:28:03] [ Info ] buffer name: 
[ SpaceVim ] [14:28:03] [ Info ] change to root: /home/max
[ SpaceVim ] [14:28:04] [ Info ] Start to find root for: /tmp/nvimcvujq3/1/issue_report.md
[ SpaceVim ] [14:28:04] [ Info ] buffer name: /tmp/nvimcvujq3/1/issue_report.md
[ SpaceVim ] [14:28:04] [ Info ] change to root: /home/max
```
</details>

**Comment by wsdjeg** [comment]
I will test it.



---

Shidong Wang
在 Max Pollack <notifications@github.com>，2020年2月1日 03:31写道：

Expected behavior, english is required

When opening a file in vim from a working directory that contains a .git folder, that folder will be treated as the project root and fzf behavior (as well as things like the runner) will be executed in the context of that working directory.

The reproduce ways from Vim starting (Required!)

Any time I open any file that's saved in any subdirectory of my home directory, it will default to the entire home directory being my default project. This makes all the project-level searching/grepping useless as it greps every file on my machine.

I see that it's changing my working directory to my home directory in the logs you asked me to post, but I really don't have insight as to why. I do not have a .git folder in my home directory, and my other machines have a .SpaceVim.d without any trouble.

Debug info
Environment Information

  *   OS: linux
  *   vim version: -
  *   neovim version: 0.4.3
  *   SpaceVim version: 1.4.0-dev
  *   SpaceVim status:

On branch master

Your branch is up to date with 'origin/master'.



nothing to commit, working tree clean



SpaceVim debug information
SpaceVim options :

  auto_disable_touchpad = 1

  autocomplete_method = 'asyncomplete'

  autocomplete_parens = 1

  buffer_index_type = 4

  checkinstall = 1

  colorscheme = 'onedark'

  colorscheme_bg = 'dark'

  colorscheme_default = 'desert'

  commandline_prompt = '➭'

  custom_color_palette = []

  custom_plugins = []

  debug_level = 1

  default_indent = 2

  disabled_plugins = []

  enable_ale = 0

  enable_bepo_layout = 0

  enable_cursorcolumn = 0

  enable_cursorline = 1

  enable_debug = 0

  enable_googlesuggest = 0

  enable_guicolors = 1

  enable_insert_leader = 1

  enable_javacomplete2_py = 0

  enable_key_frequency = 0

  enable_language_specific_leader = 1

  enable_neocomplcache = 0

  enable_neomake = 1

  enable_os_fileformat_icon = 0

  enable_powerline_fonts = 1

  enable_statusline_bfpath = 0

  enable_statusline_mode = 1

  enable_statusline_tag = 1

  enable_tabline_filetype_icon = 1

  enable_tabline_ft_icon = 0

  enable_vimfiler_filetypeicon = 0

  enable_vimfiler_gitstatus = 0

  enable_vimfiler_welcome = 1

  enable_ycm = 0

  error_symbol = '✖'

  expand_tab = 1

  filemanager = 'vimfiler'

  filetree_direction = 'right'

  filetype_icons = {}

  force_global_config = 0

  gitcommit_issue_icon = ''

  gitcommit_pr_icon = ''

  github_username = ''

  guifont = ''

  hiddenfileinfo = 1

  home_files_number = 6

  hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'

  info_symbol = 'ⓘ'

  keep_server_alive = 1

  language = ''

  leader_guide_default_group_name = ''

  leader_guide_flatten = 1

  leader_guide_hspace = 5

  leader_guide_max_size = 0

  leader_guide_position = 'botright'

  leader_guide_run_map_on_popup = 1

  leader_guide_sort_horizontal = 0

  leader_guide_submode_mappings = {'<C-C>': 'win_close'}

  leader_guide_vertical = 0

  lint_on_save = 1

  lint_on_the_fly = 0

  max_column = 120

  plugin_bundle_dir = '/home/max/.cache/vimfiles/'

  plugin_manager = 'dein'

  plugin_manager_processes = 16

  plugin_name = 'goyo.vim'

  project_rooter_automatically = 1

  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/', 'Cargo.toml', '.SpaceVim.d/']

  realtime_leader_guide = 1

  relativenumber = 0

  search_tools = ['rg', 'ag', 'pt', 'ack', 'grep', 'findstr']

  sidebar_direction = ''

  sidebar_width = 30

  smartcloseignoreft = ['tagbar', 'vimfiler', 'defx', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']

  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']

  snippet_engine = 'neosnippet'

  src_root = 'E:\sources\'

  statusline_inactive_separator = 'bar'

  statusline_iseparator = 'nil'

  statusline_left_sections = ['winnr', 'filename', 'major mode', 'search count', 'syntax checking', 'minor mode lighters']

  statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']

  statusline_separator = 'slant'

  statusline_unicode_symbols = 1

  terminal_cursor_shape = 2

  update_retry_cnt = 3

  version = '1.4.0-dev'

  vim_help_language = 'en'

  vimcompatible = 0

  warning_symbol = '⚠'

  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'

  windows_index_type = 3

  windows_leader = 's'

  windows_smartclose = 'q'

SpaceVim layers :

[[layers]]

  name="autocomplete"

  auto-completion-return-key-behavior='complete'

  auto-completion-tab-key-behavior='cycle'

[[layers]]

  name="checkers"

[[layers]]

  name="format"

[[layers]]

  name="edit"

[[layers]]

  name="ui"

[[layers]]

  name="core"

[[layers]]

  name="core#banner"

[[layers]]

  name="core#statusline"

[[layers]]

  name="core#tabline"

[[layers]]

  name="colorscheme"

[[layers]]

  name="shell"

  default_position='top'

  default_height=30

[[layers]]

  name="lang#go"

[[layers]]

  name="lang#rust"

  recommended-style=1

[[layers]]

  name="lang#ruby"

[[layers]]

  name="lsp"

[[layers]]

  name="fzf"

[[layers]]

  name="tools"

SpaceVim Health checking :

SpaceVim clipboard support check report:
Checking +clipboard:
SUCCEED!
SpaceVim environment check report:
Current progpath: nvim(/usr/bin/nvim)
version: 800
OS: linux
[shell, shellcmdflag, shellslash]: ['/usr/bin/zsh', '-c', 0]
SpaceVim lua support check report:
Checking +lua:
Failed : Known issue, neovim do not support lua now.
SpaceVim python support check report:
Checking +python3:
Failed : to support +python3, you need run pip3 install neovim
Checking +python:
Failed : to support +python, you need run pip2 install neovim

SpaceVim runtime log :

[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!

[ SpaceVim ] [14:28:02] [ Info ] Startup with no argv, current dir is used: ~/dev/rust/test_app

[ SpaceVim ] [14:28:02] [ Info ] Can not find project local config, start to loadding global config

[ SpaceVim ] [14:28:02] [ Info ] start to apply config [glob]

[ SpaceVim ] [14:28:02] [ Info ] SpaceVim server startup at:/tmp/spacevim_nvim_server

[ SpaceVim ] [14:28:03] [ Info ] Start to find root for: /home/max/dev/rust/test_app/

[ SpaceVim ] [14:28:03] [ Info ]         (.git/):/home/max/dev/rust/test_app

[ SpaceVim ] [14:28:03] [ Info ]         (Cargo.toml):/home/max/dev/rust/test_app

[ SpaceVim ] [14:28:03] [ Info ]         (.SpaceVim.d/):/home/max

[ SpaceVim ] [14:28:03] [ Info ] buffer name:

[ SpaceVim ] [14:28:03] [ Info ] change to root: /home/max

[ SpaceVim ] [14:28:03] [ Info ] try to open SpaceVim welcome page

[ SpaceVim ] [14:28:03] [ Info ] buffer name:

[ SpaceVim ] [14:28:03] [ Info ] change to root: /home/max

[ SpaceVim ] [14:28:04] [ Info ] Start to find root for: /tmp/nvimcvujq3/1/issue_report.md

[ SpaceVim ] [14:28:04] [ Info ] buffer name: /tmp/nvimcvujq3/1/issue_report.md

[ SpaceVim ] [14:28:04] [ Info ] change to root: /home/max

**Comment by wsdjeg** [comment]
This is feature of SpaceVim, it will find the root of your project automatically, and this feature is based on the `project_rooter_patterns`, 

```
[ SpaceVim ] [14:28:03] [ Info ] Start to find root for: /home/max/dev/rust/test_app/
[ SpaceVim ] [14:28:03] [ Info ]         (.git/):/home/max/dev/rust/test_app
[ SpaceVim ] [14:28:03] [ Info ]         (Cargo.toml):/home/max/dev/rust/test_app
[ SpaceVim ] [14:28:03] [ Info ]         (.SpaceVim.d/):/home/max
[ SpaceVim ] [14:28:03] [ Info ] buffer name: 
[ SpaceVim ] [14:28:03] [ Info ] change to root: /home/max
```

I just checkout the log, and the issue is spacevim change the root to home, instead of using current project root. I will fix it, this is the bug I saied in #3031

cc @Fetz
